pkg/deploy/controller/deploymentconfig/controller.go

```diff
@@ -142,6 +145,7 @@ func (c *DeploymentConfigController) Handle(config *deployapi.DeploymentConfig)
 		if !deployutil.IsTerminatedDeployment(latestDeployment) {
 			return c.updateStatus(config, existingDeployments)
 		}
+
 		return c.reconcileDeployments(existingDeployments, config)
 	}
 	// If the config is paused we shouldn't create new deployments for it.
```

Comment:
> The new deployment doesn't have to do anything with cleanup policy 

No, the *only* time a cleanup would ever have to happen is if a new deployment were made and that bumped the number of old deployments past the limit.

Although, a case where this *would* be useful is if a paused deployment is edited to add a limit in the config ... and that limit is smaller than how many old deployments currently exist. Honestly, I'm thinking we should just `defer` the cleanup block since we basically want to run it always. Groking all these edge cases is a nightmare.